enterprise/frontend/src/embedding-sdk/cli/steps/check-if-react-project.ts

```diff
@@ -21,49 +22,40 @@ export const checkIfReactProject: CliStepMethod = async state => {
 
   if (!(await hasPackageJson())) {
     spinner.fail();
-    return [
-      {
-        type: "error",
-        message: PACKAGE_JSON_NOT_FOUND_MESSAGE,
-      },
-      state,
-    ];
+    return [{ type: "error", message: PACKAGE_JSON_NOT_FOUND_MESSAGE }, state];
   }
 
-  const reactDep = await getPackageVersion("react");
-  const reactDomDep = await getPackageVersion("react-dom");
+  const dependencyVersions = await getPackageVersions("react", "react-dom");
+
+  const reactDep = dependencyVersions["react"];
+  const reactDomDep = dependencyVersions["react-dom"];
 
   const hasReactDependency = reactDep && reactDomDep;
+
   const hasSupportedReactVersion =
-    isReactVersionSupported(reactDep) && isReactVersionSupported(reactDomDep);
+    hasReactDependency &&
+    isReactVersionSupported(reactDep) &&
+    isReactVersionSupported(reactDomDep);
+
+  let warningMessage: string | null = null;
 
   if (!hasReactDependency) {
-    spinner.fail();
-    return [
-      {
-        type: "error",
-        message: MISSING_REACT_DEPENDENCY,
-      },
-      state,
-    ];
+    warningMessage = MISSING_REACT_DEPENDENCY;
+  } else if (!hasSupportedReactVersion) {
+    warningMessage = UNSUPPORTED_REACT_VERSION;
   }
 
-  if (!hasSupportedReactVersion) {
+  if (warningMessage) {
     spinner.fail();
-    return [
-      {
-        type: "error",
-        message: UNSUPPORTED_REACT_VERSION,
-      },
-      state,
-    ];
+
+    const shouldContinue = await showWarningAndAskToContinue(warningMessage);
```

Comment:
This prints the warning, and asks them if they wanted to continue or not. See the PR description for the screenshot.